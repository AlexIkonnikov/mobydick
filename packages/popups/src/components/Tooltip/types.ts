import {TextStyle, ViewStyle} from 'react-native';
import {ReactNode} from 'react';

export interface IPosition {
  top?: number | string;
  left?: number | string;
  bottom?: number | string;
  right?: number | string;
}

export interface ITooltip {
  message: string | null;
  position?: IPosition;
  element?: JSX.Element;
  styleContainer?: ViewStyle;
  textStyle?: TextStyle;
  isArrow?: boolean;
  arrowStyle?: ViewStyle;
  children?: ReactNode;
}
