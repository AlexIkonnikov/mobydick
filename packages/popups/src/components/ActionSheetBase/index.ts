import ActionSheetBase from './ActionSheetBase';

export {ActionSheetBase};
