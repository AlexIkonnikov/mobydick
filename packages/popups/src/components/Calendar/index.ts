import Calendar from './Calendar';

export {Calendar};

export * from './types';
