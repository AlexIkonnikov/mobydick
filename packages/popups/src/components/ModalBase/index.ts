import ModalBase from './ModalBase';

export {ModalBase};

export * from './types';
