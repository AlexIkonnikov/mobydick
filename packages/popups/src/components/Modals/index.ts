import ModalError from './ModalError';
import ModalLoading from './ModalLoading';
import ModalSuccess from './ModalSuccess';
import ModalCalendar from './ModalCalendar';

export {ModalError, ModalLoading, ModalSuccess, ModalCalendar};
