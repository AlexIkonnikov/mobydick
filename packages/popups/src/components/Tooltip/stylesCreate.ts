import {IUseStylesTheme} from '@npm/mobydick-styles';
import {StyleSheet} from 'react-native';

const stylesCreate = (theme: IUseStylesTheme) => {
  const {colors} = theme;
  return StyleSheet.create({
    overlayStyle: {
      backgroundColor: 'transparent',
    },
    container: {
      paddingVertical: 8,
      paddingHorizontal: 12,
      borderRadius: 8,
      justifyContent: 'center',
      alignItems: 'center',
      zIndex: 1,

      backgroundColor: colors.IconFavorite,
    },
    arrow: {
      position: 'absolute',
      borderRadius: 4,
      width: 0,
      height: 0,
      backgroundColor: 'transparent',
      borderStyle: 'solid',
      //позиция стрелочки меняется в зависимости этих стилей
      right: 10,
      bottom: -7,
      borderTopWidth: 8,
      borderRightWidth: 10,
      borderBottomWidth: 0,
      borderLeftWidth: 10,
      borderTopColor: colors.IconFavorite,
      borderRightColor: 'transparent',
      borderBottomColor: 'transparent',
      borderLeftColor: 'transparent',
    },
    text: {
      fontSize: 14,
      zIndex: 1,
      color: colors.TextPrimary,
    },
  });
};

export default stylesCreate;
