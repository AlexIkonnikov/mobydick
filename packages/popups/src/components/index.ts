export * from './PopupBase';
export * from './ModalBase';
export * from './SnackbarBase';
export * from './TooltipBase';
export * from './ActionSheetBase';
