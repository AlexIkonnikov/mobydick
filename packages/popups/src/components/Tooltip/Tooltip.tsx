import React, {FC} from 'react';
import {Animated} from 'react-native';
import {useStyles} from '@npm/mobydick-styles';
import {Text, View} from '@npm/mobydick-core';
import {ITooltip} from '@npm/mobydick-popups/src/components/Tooltip/types';

import stylesCreate from './stylesCreate';

const Tooltip: FC<ITooltip> = props => {
  const {
    message,
    position,
    styleContainer,
    arrowStyle,
    textStyle,
    isArrow,
    children,
  } = props;
  const [styles] = useStyles(stylesCreate);

  return (
    <Animated.View
      style={[
        styles.container,
        styleContainer,
        position && {
          position: 'absolute',
          top: position?.top,
          bottom: position?.bottom,
          left: position?.left,
          right: position?.right,
        },
      ]}>
      <Text style={[styles.text, textStyle]}>{message}</Text>
      {isArrow && <View style={[styles.arrow, arrowStyle]} />}
      {children}
    </Animated.View>
  );
};

export default Tooltip;
