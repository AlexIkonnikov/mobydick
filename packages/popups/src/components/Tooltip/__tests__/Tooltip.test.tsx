import {render} from '@testing-library/react-native';
import React from 'react';
import {Tooltip} from '@npm/mobydick-popups/src/components/Tooltip';

describe('@npm/mobydick-popups/Tooltip', () => {
  it('should renders correctly', () => {
    const {toJSON} = render(
      <Tooltip
        message={'message'}
        position={{top: '120', bottom: '10', left: '10', right: '10'}}
        isArrow
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
