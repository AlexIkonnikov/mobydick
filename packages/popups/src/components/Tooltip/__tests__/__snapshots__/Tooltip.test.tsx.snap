// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/Tooltip should renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FAB742",
      "borderRadius": 8,
      "bottom": "10",
      "justifyContent": "center",
      "left": "10",
      "paddingHorizontal": 12,
      "paddingVertical": 8,
      "position": "absolute",
      "right": "10",
      "top": "120",
      "zIndex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#20242D",
          "fontSize": 14,
          "zIndex": 1,
        },
        undefined,
      ]
    }
  >
    message
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "transparent",
          "borderBottomWidth": 0,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 10,
          "borderRadius": 4,
          "borderRightColor": "transparent",
          "borderRightWidth": 10,
          "borderStyle": "solid",
          "borderTopColor": "#FAB742",
          "borderTopWidth": 8,
          "bottom": -7,
          "height": 0,
          "position": "absolute",
          "right": 10,
          "width": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;
